src/main/kotlin/graphics/scenery/backends/vulkan/VulkanRenderer.kt

```diff
@@ -709,6 +711,17 @@ open class VulkanRenderer(hub: Hub,
                 usage = VK_BUFFER_USAGE_VERTEX_BUFFER_BIT or VK_BUFFER_USAGE_INDEX_BUFFER_BIT or VK_BUFFER_USAGE_TRANSFER_DST_BIT
             )
 
+            ssboUploadPool = VulkanBufferPool(
+                device,
+                usage = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT or VK_BUFFER_USAGE_TRANSFER_DST_BIT
```

Comment:
Just to double-check: the upload pool is intended to receive data from the CPU side, yes? Then `TRANSFER_DST_BIT` is correct. Otherwise the two need to be switched. Also, is there a particular reason to have two pools? There's likely no performance issue with having one that has both `SRC` and `DST`.

Reply:
On the GPU side, there is a slight performance boost, as SSBOs are layed out differently. Also, a read only buffer can live in faster device local memory, while a writable buffer needs to be host visible or coherent (not to familiar with the concepts yet)
Thats why I decided to split up and download buffers into 2. You dont always want to write into a buffer and vice verse. Especially for read only buffers it might be good to keep this practice